code/modules/mob/living/simple_animal/hostile/zombie.dm

```diff
@@ -169,34 +176,84 @@ mob/living/simple_animal/hostile/zombie/gymboss
 	desc = "You probably shouldn't use this particular shovel."
 	var/used = FALSE
 	force = 15
+	var/pit_type = /mob/living/simple_animal/hostile/spawner/zombie
 
 /obj/item/weapon/shovel/cursed/attack_self(mob/living/user)
 	if(used == FALSE)
 		visible_message("<span class='danger'>[user] digs a pit to Hell!</span>")
-		new /mob/living/simple_animal/hostile/spawner/zombie(get_turf(src.loc))
+		new pit_type(get_turf(src.loc))
 		used = TRUE
 	else
 		user << "The unearthly energies that once powered this shovel are now dormant. Still sharp though."
 
+/obj/item/organ/body_egg/zombie_infection
+	name = "festering ooze"
+	desc = "A black web of pus and vicera."
+	zone = "head"
+	slot = "zombie_infection"
+	var/reanimation_time
+	var/list/faction
 
-/mob/living/simple_animal/hostile/zombie/holder
-	name = "infection holder"
-	icon_state = "none"
-	icon_living = "none"
-	icon_dead = "none"
-	desc = "You shouldn't be seeing this."
-	invisibility = INVISIBILITY_ABSTRACT
-	unsuitable_atmos_damage = 0
-	stat_attack = 2
-	gold_core_spawnable = 0
-	AIStatus = AI_OFF
-	stop_automated_movement = 1
-	density = 0
+/obj/item/organ/body_egg/zombie_infection/New()
+	. = ..()
+	faction = list("zombie")
 
-/mob/living/simple_animal/hostile/zombie/holder/New()
-	..()
-	spawn(rand(800,1200))
-		if(src && istype(loc, /mob/living/carbon/human))
-			var/mob/living/carbon/human/H = loc
-			Zombify(H)
-		qdel(src)
+/obj/item/organ/body_egg/zombie_infection/on_find(mob/living/finder)
+	finder << "<span class='warning'>Inside the head is a disgusting black \
+		web of pus and vicera, bound tightly around the brain like some \
+		biological harness.</span>"
+
+/obj/item/organ/body_egg/zombie_infection/egg_process()
+	if(!ishuman(owner)) // We do not support monkey or xeno zombies
+		return
+	if(owner.stat != DEAD)
+		reanimation_time = null
+		return
+	if(istype(owner.loc, /mob/living/simple_animal/hostile/zombie))
+		reanimation_time = null
+		return
+
+	if(!reanimation_time)
+		reanimation_time = world.time + rand(800, 1200)
+		return
+	if(reanimation_time < world.time)
+		reanimation_time = null
+		zombify() // Rise and shine, Mr Romero... rise and shine.
+
+/obj/item/organ/body_egg/zombie_infection/proc/zombify()
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/H = owner
+	H.set_species(/datum/species/zombie)
+
+	if(H.head) //So people can see they're a zombie
+		var/obj/item/clothing/helmet = H.head
+		if(!H.unEquip(helmet))
+			qdel(helmet)
+	if(H.wear_mask)
+		var/obj/item/clothing/mask = H.wear_mask
+		if(!H.unEquip(mask))
+			qdel(mask)
+	var/mob/living/simple_animal/hostile/zombie/Z = new(H.loc)
+
+	Z.faction = faction.Copy()
+	Z.appearance = H.appearance
+	Z.transform = matrix()
```

Comment:
resets the matrix to default (Identity matrix).
the thing is, Z was only just created, so it's matrix *IS* already default so this line does nothing.